test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -129,43 +130,27 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 	})
 
 	It("Should create, upgrade and delete a workload cluster", func() {
-		By("Deploy Test Extension")
-		testExtensionDeploymentTemplate, err := os.ReadFile(testExtensionPath) //nolint:gosec
-		Expect(err).ToNot(HaveOccurred(), "Failed to read the extension deployment manifest file")
-
-		// Set the TEST_EXTENSION_SERVICE_NAMESPACE, which is used in the cert-manager Certificate CR.
-		// We have to dynamically set the namespace here, because it depends on the test run and thus
-		// cannot be set when rendering the test extension YAML with kustomize.
-		testExtensionDeployment := strings.ReplaceAll(string(testExtensionDeploymentTemplate), "${TEST_EXTENSION_SERVICE_NAMESPACE}", namespace.Name)
-
-		Expect(testExtensionDeployment).ToNot(BeEmpty(), "Test Extension deployment manifest file should not be empty")
-		Expect(input.BootstrapClusterProxy.Apply(ctx, []byte(testExtensionDeployment), "--namespace", namespace.Name)).To(Succeed())
+		// NOTE: test extension is already deployed in the management cluster. If for any reason in future we want
+		// to make this test more self-contained this test should be modified in order to create a management cluster
+		// to be initialized with the management cluster; also the E2E test configuration should be modified introducing
+		// something like option:true allowing to define which providers should not be installed by default in
+		// a management cluster.
```

Comment:
"create a management cluster to be initialized with the management cluster"
This sounds a bit confusing

nit: I guess `option:true` > `optional:true`